CMakeLists.txt

```diff
@@ -9,6 +9,11 @@ project(VUL LANGUAGES CXX)
 
 string(COMPARE EQUAL ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_SOURCE_DIR} VUL_IS_TOP_LEVEL) # Remove when min is 3.21
 
+# Control whether to install or not. By default its only on if this is the top level cmake project.
+if(NOT DEFINED VUL_ENABLE_INSTALL)
```

Comment:
Why is an if statement put around the install code? Shouldn't making the install option be FALSE if VUL_IS_TOP_LEVEL is false be sufficient?

Put another way, we do not need to guard against the variable not existing before defining it. `option()` does not override the value if the variable already exists.
https://cmake.org/cmake/help/latest/command/option.html

Reply:
Good Question: If I recall correctly, its mainly to make the behavior consistent with older cmake versions: https://cmake.org/cmake/help/latest/policy/CMP0077.html#policy:CMP0077

I am not married to that check, if you think its too ugly, feel free to remove it!